esp/esp/program/modules/forms/resources.py

```diff
@@ -61,9 +61,16 @@ def save_restype(self, program, res_type):
             res_type.program = program
         if self.cleaned_data['priority']:
             res_type.priority_default = self.cleaned_data['priority']
+        if choices and filter(None, choices):
+            res_type.choices = filter(None, choices)
+        else:
+            res_type.attributes_pickled = ResourceType._meta.get_field('attributes_pickled').default
```

Comment:
Do you need this?  I would think it would set the default automatically here.  (Also, if you do, may want to comment that setting `choices` and `attributes_pickled` is equivalent.)

Reply:
Hmm...you may be correct. I'll look into it.

Reply:
After some testing, I recalled that this is in case you remove all of the choices from a pre-existing resource type. I've added a comment explaining that and your note as well.